src/OrbitGl/App.h

```diff
@@ -145,6 +145,14 @@ class OrbitApp final : public DataViewFactory, public CaptureListener {
   void OnValidateFramePointers(std::vector<const ModuleData*> modules_to_validate);
 
   void SetCaptureWindow(CaptureWindow* capture);
+  [[nodiscard]] const TimeGraph* GetTimeGraph() const {
+    CHECK(capture_window_ != nullptr);
+    return capture_window_->GetTimeGraph();
+  }
+  [[nodiscard]] TimeGraph* GetMutableTimeGraph() {
+    CHECK(capture_window_ != nullptr);
+    return capture_window_->GetTimeGraph();
+  }
```

Comment:
There is an `if (capture_window != nullptr)` somewhere in the code. That tells me that we can't assume that `capture_window_` is always existing. Have you considered returning a `std::optional<TimeGraph*>` and not abort if `capture_window_` does not exist?

Reply:
Well, yeah I though of this. But in the code there is exactly one place where we do the capture_window != nullptr check, on all other places we currently assume that GCurrentTimeGraph is not null (actually without stating it explicitly). I am not sure if we want to add handling for " no capture window present" for all the callers, or if we would end up in adding CHECKs on every caller, in which case we could also just keep it as it is now. It is an improvement to the current state, as we at least do have the checks here.

But I am happy to discuss, and don't have strong feelings about this.

Nit question: Assume we do handle this case here, why not just returning nullptr?

Reply:
I see your point. Before we ignored it, but it wasn't so obvious. Now, we keep the ignoring part but at least improve in other parts. So I guess it's fine for me to leave it.

Regarding nullptr: I prefer not to return a nullptr, but rather make a syntactical change. We had a lot of crashes over the last few releases which were related to similar getters. `GetProcess`, `GetProcessManager`, `GetModuleByPath`. You never know whether you can assume the returned pointer to be valid or not. And someone might change the behaviour and it will still compile. That's why I advocate for a raw pointer to be never null. If you need an empty value, wrap it into a std::optional.

I want to make a small presentation about that because I feel that is related to a lot of crashes in the past. But of course that shouldn't stop you now. :-)

Reply:
Okay, then lets leave it for now.